Tests/KeyHasherSpec.swift

```diff
@@ -0,0 +1,126 @@
+import Quick
+import Nimble
+@testable import RInAppMessaging
+
+class KeyHasherSpec: QuickSpec {
+
+    override func spec() {
+        // Be sure to run those tests for ios version >=13.0 and <13.0
+        describe("KeyHasher") {
+
+            var hasher: KeyHasher!
+
+            beforeEach {
+                hasher = KeyHasher()
+            }
+
+            [KeyHasher.Encryption.md5, KeyHasher.Encryption.sha256].forEach { encryption in
+                context("when using \(encryption) encryption method") {
+
+                    beforeEach {
+                        hasher.encryprtionMethod = encryption
+                    }
+
+                    it("will generate a valid hash for empty data") {
+                        expect(hasher.generateHash()).toNot(beEmpty())
+                    }
+
+                    it("will generate a valid hash for empty value") {
```

Comment:
Is this the anonymous (no user ids) case?

Reply:
hmm I think it's rather the "no data" one.